File: mobile/src/types/quote.ts
export interface QuoteItem {
  id: string
  label: string
  category: string | null
  quantity: number
  unit_price_cents: number | null
  total_price_cents: number | null
}

export interface Quote {
  id: string
  deal_id: string
  document_id: string | null
  supplier_name: string | null
  quote_number: string | null
  amount_excl_tax_cents: number | null
  amount_incl_tax_cents: number | null
  currency: string
  category: string | null
  extraction_status: 'pending' | 'done' | 'failed'
  items: QuoteItem[]
  created_at: string
  updated_at: string
}

export interface QuoteCreatePayload {
  document_id?: string | null
  supplier_name?: string | null
  quote_number?: string | null
  amount_excl_tax_cents?: number | null
  amount_incl_tax_cents?: number | null
  currency?: string
  category?: string | null
}

File: backend/app/models/quote.py
import uuid
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, ForeignKey, Integer, String, func
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.core.db import Base


class Quote(Base):
    __tablename__ = "quotes"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    deal_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("deals.id", ondelete="CASCADE"), nullable=False)
    document_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), ForeignKey("documents.id", ondelete="SET NULL"), nullable=True)
    supplier_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    quote_number: Mapped[str | None] = mapped_column(String(100), nullable=True)
    amount_excl_tax_cents: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    amount_incl_tax_cents: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    currency: Mapped[str] = mapped_column(String(3), nullable=False, server_default="EUR")
    category: Mapped[str | None] = mapped_column(String(100), nullable=True)
    extraction_status: Mapped[str] = mapped_column(String(50), nullable=False, server_default="pending")
    extraction_payload: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)


class QuoteItem(Base):
    __tablename__ = "quote_items"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    quote_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("quotes.id", ondelete="CASCADE"), nullable=False)
    label: Mapped[str] = mapped_column(String(255), nullable=False)
    category: Mapped[str | None] = mapped_column(String(100), nullable=True)
    quantity: Mapped[int] = mapped_column(Integer, nullable=False, server_default="1")
    unit_price_cents: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    total_price_cents: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
